packages/server/src/integrations/utils.ts

```diff
@@ -440,3 +445,59 @@ export function removeEmptyFilters(filters: SearchFilters) {
   }
   return filters
 }
+
+export async function handleXml(response: any) {
+  let data
+  const rawXml = await response.text()
+  data =
+    (await xmlParser(rawXml, {
+      explicitArray: false,
+      trim: true,
+      explicitRoot: false,
+    })) || {}
+  // there is only one structure, its an array, return the array so it appears as rows
+  const keys = Object.keys(data)
+  if (keys.length === 1 && Array.isArray(data[keys[0]])) {
+    data = data[keys[0]]
+  }
+  return { data, rawXml }
+}
+
+export async function handleFileResponse(
+  response: any,
+  filename: string,
+  startTime: number
+) {
+  let presignedUrl
+  const responseBuffer = await response.arrayBuffer()
+  const fileExtension = filename.includes(".")
```

Comment:
`filename` _could_ include directory information, and directory names _could_ include dots. I think we probably want to either reject directories, strip them, or handle them somehow in this check.